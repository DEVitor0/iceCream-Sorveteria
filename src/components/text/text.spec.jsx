import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Text from './Text';

describe('Text Component', () => {
  it('renders the text correctly', () => {
    render(<Text text="Sample Text" fontSize={18} />);
    const paragraph = screen.getByText('Sample Text');
    expect(paragraph).toBeInTheDocument();
  });

  it('applies the correct font size', () => {
    render(<Text text="Sample Text" fontSize={24} />);
    const paragraph = screen.getByText('Sample Text');
    expect(paragraph).toHaveStyle('font-size: 24px');
  });

  it('applies the class name from imported styles', () => {
    render(<Text text="Sample Text" fontSize={18} />);
    const paragraph = screen.getByText('Sample Text');
    expect(paragraph).toHaveClass('paragraph');
  });
});
