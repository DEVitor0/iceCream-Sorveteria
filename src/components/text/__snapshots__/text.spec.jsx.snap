// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component applies the class name from imported styles 1`] = `
<DocumentFragment>
  <p
    class="paragraph "
    style="font-size: 18px;"
  >
    Sample Text
  </p>
</DocumentFragment>
`;

exports[`Text Component applies the correct font size 1`] = `
<DocumentFragment>
  <p
    class="paragraph "
    style="font-size: 24px;"
  >
    Sample Text
  </p>
</DocumentFragment>
`;

exports[`Text Component renders the text correctly 1`] = `
<DocumentFragment>
  <p
    class="paragraph "
    style="font-size: 18px;"
  >
    Sample Text
  </p>
</DocumentFragment>
`;
